server/example_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"gocloud.dev/health"
+	"gocloud.dev/requestlog"
+	"gocloud.dev/server"
+)
+
+func ExampleServer_New() {
+	// This example is used in https://gocloud.dev/howto/server/
+
+	// Use the constructor function to create the server.
+	srv := server.New(http.DefaultServeMux, nil)
+
+	// Register a route.
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "Hello, World!")
+	})
+
+	// Start the server.
+	srv.ListenAndServe(":8080")
+}
+
+func ExampleServer_RequestLogger() {
+	// This example is used in https://gocloud.dev/howto/server/
+
+	// Create a logger, and assign it to the RequestLogger field of a
+	// server.Options struct.
+	srvOptions := &server.Options{
+		RequestLogger: requestlog.NewNCSALogger(os.Stdout, func(error) {}),
+	}
+
+	// Pass the options to the Server constructor.
+	srv := server.New(http.DefaultServeMux, srvOptions)
+
+	// Register a route.
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "Hello, World!")
+	})
+
+	// Start the server. You will see requests logged to STDOUT.
+	if err := srv.ListenAndServe(":8080"); err != nil {
+		fmt.Fprintln(os.Stderr, err)
```

Comment:
Consider using `log.Exitf` instead of `fmt.Fprint` and `os.Exit`.

Reply:
this one? https://godoc.org/github.com/golang/glog#Exitf  

(https://godoc.org/log doesn't seem to have anything called `Exitf`?)

Reply:
Sorry, `Fatalf`: https://golang.org/pkg/log/#Fatalf

Reply:
FWIW, we're using `log.Fatal` for this in other examples. Let's use the same here? There's also `log.Fatalf` if needed (not here)